src/hazelcore/Statics_Exp.re

```diff
@@ -2204,7 +2203,14 @@ module rec M: Statics_Exp_Sig.S = {
     | FreeLivelit(_) =>
       let (e, ty', u_gen) =
         syn_fix_holes_operand(ctx, u_gen, ~renumber_empty_holes, e);
-      if (HTyp.consistent(ty, ty')) {
+      let is_insufficient_params =
+        switch (UHExp.get_err_status_operand(e)) {
+        | InHole(TypeInconsistent(Some(InsufficientParams)), _) => false
+        | _ => true
+        };
+      if (is_insufficient_params) {
+        (e, u_gen);
+      } else if (HTyp.consistent(ty, ty')) {
         (UHExp.set_err_status_operand(NotInHole, e), u_gen);
```

Comment:
I don't think we actually need this branch (in dev too), because syn_fix_holes should never produce a term in a type inconsistency hole

Reply:
Which branch? The middle one? If I remove it, the image example immediately throws an error on-load:

"Async_kernel: Unhandled exception (Failure
  "Unexpected evaluation failure during livelit args fill and resume: 1")"